plugins/src/test/java/opengrok/auth/plugin/UserWhiteListPluginTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020, Chris Fraire <cfraire@me.com>.
+ */
+
+package opengrok.auth.plugin;
+
+import opengrok.auth.plugin.entity.User;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.opengrok.indexer.configuration.Group;
+import org.opengrok.indexer.configuration.Project;
+import org.opengrok.indexer.util.RandomString;
+import org.opengrok.indexer.web.DummyHttpServletRequest;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Represents a container for tests of {@link UserWhiteListPlugin}.
+ */
+public class UserWhiteListPluginTest {
+
+    private static final String OK_USER = "user1321";
+    private static File tempWhitelist;
+    private static HashMap<String, Object> validPluginParameters;
+
+    private UserWhiteListPlugin plugin;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        tempWhitelist = File.createTempFile("UserWhiteListPluginTest", "txt");
+        try (BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+                new FileOutputStream(tempWhitelist), StandardCharsets.UTF_8))) {
+            writer.write(OK_USER);
+            // Don't bother with trailing LF.
+        }
+
+        validPluginParameters = new HashMap<>();
+        validPluginParameters.put(UserWhiteListPlugin.FILE_PARAM, tempWhitelist.getPath());
```

Comment:
would be nice to see what happens if the parameters include path to non-existent file.

Reply:
Done